pkg/virt-handler/device-manager/device_controller.go

```diff
@@ -62,73 +59,26 @@ func (c *DeviceController) nodeHasDevice(devicePath string) bool {
 	return (err == nil)
 }
 
-func (c *DeviceController) waitForPath(target string, stop chan struct{}) error {
+func (c *DeviceController) startDevicePlugin(dev GenericDevice, stop chan struct{}) {
 	logger := log.DefaultLogger()
+	deviceName := dev.GetDeviceName()
 
-	_, err := os.Stat(target)
-	if err != nil {
-		if !os.IsNotExist(err) {
-			return err
+	for {
+		err := dev.Start(stop)
```

Comment:
do we want to re-register device each second, why we can not run `health check` before and if it returns ok, just continue without re-register?

Reply:
It tries every second if it can't register with the kubelet. I could exchange that with a back-off, so taht we don't pollute the logs too much. Blind retries might not be the most elegant solution, but even if the kubelet socket file is there we might not be able to connect (e.g. kubelet crashed and is not running).
 
In all other cases it stays connected and only switches the devices to healthy/unhealthy. That is different to before, where it waited until the device showed up and had similar checks also inside the plugin. I collapsed that to one health check.

Reply:
hm if kubelet crashed re-register will not help until kubelet is back, and I believe when it will back it will re-create socket or do I miss something?

Reply:
Yes, it re-creates it, but I am not sure on how to distinguish a kubelet restart or crash from other connection errors. There might be various reasons why we have a disconnect. In all these cases,  the socket file stays there and we would not re-connect.

Reply:
@cynepco3hahue added the back-off logic.